File: Data Structures & Algorithms/anagram-groups/submission-3.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */

    groupAnagrams(strs) {
       let map={}
       for(let str of strs){
            let count=Array(26).fill(0)
            for(let char of str){
                count[char.charCodeAt(0)-'a'.charCodeAt(0)]+=1
            }
            count=count.join('#')
            console.log(count)
            if(map[count]===undefined) map[count]=[str]
            else map[count].push(str)
       }
       return Object.values(map)
    }
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-3.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        res=[str(len(s))+"#"+s for s in strs]
        return "".join(res)


    def decode(self, s: str) -> List[str]:
        res=[]
        i=0
        while i<len(s):
            j=i
            while s[j]!="#":
                j+=1
            l=int(s[i:j])
            
            res.append(s[j+1:j +1+l])
            i=j+l+1
        return res